### PR TITLE
Make the project compatible with Decoder 0.7, add a `make format` command

### PR DESCRIPTION
**Pull request by hesterjeng**
This PR mostly consists of small changes to a pair of mli files to make the project compatible with `decoders` 0.7 and adds a command to the makefile to automatically format everything

### CODE REVIEW

File: src/api.mli [hunk 1]
```diff
@@ -150,17 +150,20 @@ module Decoders : functor (D : Decoders.Decode.S) -> sig
   module Request : sig
     module Hints : sig
       module Induct : sig
-        val structural : (D.value, Request.Hints.Induct.t) D.t_let
+        (* val structural : (D.value, Request.Hints.Induct.t) D.t_let *)
+        val structural : Request.Hints.Induct.t D.decoder
 
-        val functional : (D.value, Request.Hints.Induct.t) D.t_let
+        (* val functional : (D.value, Request.Hints.Induct.t) D.t_let *)
```
Comment (benbellick) [inline src/api.mli:156]: Any reason to leave the old code in as comments? If not, can we just get rid of it?

Comment (hesterjeng) [reply src/api.mli]: Done

**Comment by benbellick** [comment]
I just called it locally to make sure it worked on my computer and pushed the suggested format changes!

**Review by benbellick — APPROVED** [review]

**Comment by aidmandorky** [comment]
The project was already compatible with Decoders v.1.0.0, no? Presumably the `.mli` didn't account for the recent changes in the `t_let` stuff. LGTM anyway.

**Review by aidmandorky — APPROVED** [review]

**Pull request merged**